File: HFSGuardaDiretorio_Gtk2_C-vs2015/HFSGuardaDiretorio/HFSGuardaDiretorio/MenuPopup.c
#include "MenuPopup.h"
#include "hfsguardadir/comum/Rotinas.h"
// ---------------------------------------------------------------------------
GtkWidget *MenuPopup_Criar()
{
	MenuPopup.menuPopup = gtk_menu_new();

	MenuPopup.menuExcluirAbaAtiva2 = gtk_menu_item_new();
	gtk_widget_set_name(GTK_WIDGET(MenuPopup.menuExcluirAbaAtiva2), "menuExcluirAbaAtiva2");
	gtk_widget_set_visible(MenuPopup.menuExcluirAbaAtiva2, TRUE);
	gtk_widget_set_can_focus(MenuPopup.menuExcluirAbaAtiva2, FALSE);
	gtk_menu_item_set_label(GTK_MENU_ITEM(MenuPopup.menuExcluirAbaAtiva2), "Excluir aba ativa");
	gtk_menu_item_set_use_underline(GTK_MENU_ITEM(MenuPopup.menuExcluirAbaAtiva2), TRUE);
	
	MenuPopup.menuAlterarNomeAbaAtiva2 = gtk_menu_item_new();
	gtk_widget_set_name(GTK_WIDGET(MenuPopup.menuAlterarNomeAbaAtiva2), "menuAlterarNomeAbaAtiva2");
	gtk_widget_set_visible(MenuPopup.menuAlterarNomeAbaAtiva2, TRUE);
	gtk_widget_set_can_focus(MenuPopup.menuAlterarNomeAbaAtiva2, FALSE);
	gtk_menu_item_set_label(GTK_MENU_ITEM(MenuPopup.menuAlterarNomeAbaAtiva2), "Alterar nome da aba ativa");
	gtk_menu_item_set_use_underline(GTK_MENU_ITEM(MenuPopup.menuAlterarNomeAbaAtiva2), TRUE);
	
	MenuPopup.menuIncluirNovaAba2 = gtk_menu_item_new();
	gtk_widget_set_name(GTK_WIDGET(MenuPopup.menuIncluirNovaAba2), "menuIncluirNovaAba2");
	gtk_widget_set_visible(MenuPopup.menuIncluirNovaAba2, TRUE);
	gtk_widget_set_can_focus(MenuPopup.menuIncluirNovaAba2, FALSE);
	gtk_menu_item_set_label(GTK_MENU_ITEM(MenuPopup.menuIncluirNovaAba2), "Incluir nova aba");
	gtk_menu_item_set_use_underline(GTK_MENU_ITEM(MenuPopup.menuIncluirNovaAba2), TRUE);
	
	MenuPopup.separador2 = gtk_separator_menu_item_new();
	gtk_widget_set_name(GTK_WIDGET(MenuPopup.separador2), "separador2");
	gtk_widget_set_visible(MenuPopup.separador2, TRUE);
	gtk_widget_set_can_focus(MenuPopup.separador2, FALSE);
	
	MenuPopup.menuColapsarDiretorios2 = gtk_menu_item_new();
	gtk_widget_set_name(GTK_WIDGET(MenuPopup.menuColapsarDiretorios2), "menuColapsarDiretorios2");
	gtk_widget_set_visible(MenuPopup.menuColapsarDiretorios2, TRUE);
	gtk_widget_set_can_focus(MenuPopup.menuColapsarDiretorios2, FALSE);
	gtk_menu_item_set_label(GTK_MENU_ITEM(MenuPopup.menuColapsarDiretorios2), "Colapsar Diretórios");
	gtk_menu_item_set_use_underline(GTK_MENU_ITEM(MenuPopup.menuColapsarDiretorios2), TRUE);
	
	MenuPopup.menuExpandirDiretorios2 = gtk_menu_item_new();
	gtk_widget_set_name(GTK_WIDGET(MenuPopup.menuExpandirDiretorios2), "menuExpandirDiretorios2");
	gtk_widget_set_visible(MenuPopup.menuExpandirDiretorios2, TRUE);
	gtk_widget_set_can_focus(MenuPopup.menuExpandirDiretorios2, FALSE);
	gtk_menu_item_set_label(GTK_MENU_ITEM(MenuPopup.menuExpandirDiretorios2), "Expandir Diretórios");
	gtk_menu_item_set_use_underline(GTK_MENU_ITEM(MenuPopup.menuExpandirDiretorios2), TRUE);
	
	MenuPopup.menuExcluirDiretorioSelecionado = gtk_menu_item_new();
	gtk_widget_set_name(GTK_WIDGET(MenuPopup.menuExcluirDiretorioSelecionado), "menuExcluirDiretorioSelecionado");
	gtk_widget_set_visible(MenuPopup.menuExcluirDiretorioSelecionado, TRUE);
	gtk_widget_set_can_focus(MenuPopup.menuExcluirDiretorioSelecionado, FALSE);
	gtk_menu_item_set_label(GTK_MENU_ITEM(MenuPopup.menuExcluirDiretorioSelecionado), "Excluir Diretório Selecionado");
	gtk_menu_item_set_use_underline(GTK_MENU_ITEM(MenuPopup.menuExcluirDiretorioSelecionado), TRUE);
	
	MenuPopup.menuInformacoesDiretorioArquivo = gtk_menu_item_new();
	gtk_widget_set_name(GTK_WIDGET(MenuPopup.menuInformacoesDiretorioArquivo), "menuInformacoesDiretorioArquivo");
	gtk_widget_set_visible(MenuPopup.menuInformacoesDiretorioArquivo, TRUE);
	gtk_widget_set_can_focus(MenuPopup.menuInformacoesDiretorioArquivo, FALSE);
	gtk_menu_item_set_label(GTK_MENU_ITEM(MenuPopup.menuInformacoesDiretorioArquivo), "Informações do Diretório / Arquivo");
	gtk_menu_item_set_use_underline(GTK_MENU_ITEM(MenuPopup.menuInformacoesDiretorioArquivo), TRUE);
	
	gtk_widget_set_name(MenuPopup.menuPopup, "menuPopup");
	gtk_widget_set_visible(MenuPopup.menuPopup, TRUE);
	gtk_widget_set_can_focus(MenuPopup.menuPopup, FALSE);
	gtk_container_add(GTK_CONTAINER(MenuPopup.menuPopup), GTK_WIDGET(MenuPopup.menuInformacoesDiretorioArquivo));
	gtk_container_add(GTK_CONTAINER(MenuPopup.menuPopup), GTK_WIDGET(MenuPopup.menuExcluirDiretorioSelecionado));
	gtk_container_add(GTK_CONTAINER(MenuPopup.menuPopup), GTK_WIDGET(MenuPopup.menuExpandirDiretorios2));
	gtk_container_add(GTK_CONTAINER(MenuPopup.menuPopup), GTK_WIDGET(MenuPopup.menuColapsarDiretorios2));
	gtk_container_add(GTK_CONTAINER(MenuPopup.menuPopup), GTK_WIDGET(MenuPopup.separador2));
	gtk_container_add(GTK_CONTAINER(MenuPopup.menuPopup), GTK_WIDGET(MenuPopup.menuIncluirNovaAba2));
	gtk_container_add(GTK_CONTAINER(MenuPopup.menuPopup), GTK_WIDGET(MenuPopup.menuAlterarNomeAbaAtiva2));
	gtk_container_add(GTK_CONTAINER(MenuPopup.menuPopup), GTK_WIDGET(MenuPopup.menuExcluirAbaAtiva2));
	

	g_signal_connect((gpointer) MenuPopup.menuInformacoesDiretorioArquivo, "activate", G_CALLBACK(on_menuInformacoesDiretorioArquivo_activate), NULL); 
	g_signal_connect((gpointer) MenuPopup.menuExcluirDiretorioSelecionado, "activate", G_CALLBACK(on_menuExcluirDiretorioSelecionado_activate), NULL); 
	g_signal_connect((gpointer) MenuPopup.menuExpandirDiretorios2, "activate", G_CALLBACK(on_menuExpandirDiretorios2_activate), NULL); 
	g_signal_connect((gpointer) MenuPopup.menuColapsarDiretorios2, "activate", G_CALLBACK(on_menuColapsarDiretorios2_activate), NULL); 
	g_signal_connect((gpointer) MenuPopup.menuIncluirNovaAba2, "activate", G_CALLBACK(on_menuIncluirNovaAba2_activate), NULL); 
	g_signal_connect((gpointer) MenuPopup.menuAlterarNomeAbaAtiva2, "activate", G_CALLBACK(on_menuAlterarNomeAbaAtiva2_activate), NULL); 
	g_signal_connect((gpointer) MenuPopup.menuExcluirAbaAtiva2, "activate", G_CALLBACK(on_menuExcluirAbaAtiva2_activate), NULL); 

		
	g_signal_connect ((gpointer) MenuPopup.menuPopup, "destroy", 
		G_CALLBACK (on_MenuPopup_destroy), NULL);

	return MenuPopup.menuPopup;
}
// ---------------------------------------------------------------------------
void MenuPopup_Mostrar() 
{
    
}
// ---------------------------------------------------------------------------
void on_MenuPopup_destroy(GtkObject *object, gpointer user_data) {
	
}
// ---------------------------------------------------------------------------
void on_menuInformacoesDiretorioArquivo_activate(GtkObject *object, gpointer user_data){

}

void on_menuExcluirDiretorioSelecionado_activate(GtkObject *object, gpointer user_data){
		
}

void on_menuExpandirDiretorios2_activate(GtkObject *object, gpointer user_data){
		
}

void on_menuColapsarDiretorios2_activate(GtkObject *object, gpointer user_data){
		
}

void on_menuIncluirNovaAba2_activate(GtkObject *object, gpointer user_data){
		
}

void on_menuAlterarNomeAbaAtiva2_activate(GtkObject *object, gpointer user_data){
		
}

void on_menuExcluirAbaAtiva2_activate(GtkObject *object, gpointer user_data){
		
}


// ---------------------------------------------------------------------------

File: HFSGuardaDiretorio_Gtk2_java/src/hfsguardadiretorio/gui/MenuPopup.java
package hfsguardadiretorio.gui;

import org.gnome.gtk.Menu;
import org.gnome.gtk.MenuItem;
import org.gnome.gtk.SeparatorMenuItem;

public class MenuPopup extends Menu
{
	private MenuItem menuInformacoesDiretorioArquivo;
	private MenuItem menuExcluirDiretorioSelecionado;
	private MenuItem menuExpandirDiretorios2;
	private MenuItem menuColapsarDiretorios2;
	private SeparatorMenuItem separador2;
	private MenuItem menuIncluirNovaAba2;
	private MenuItem menuAlterarNomeAbaAtiva2;
	private MenuItem menuExcluirAbaAtiva2;


	public MenuPopup()
	{
		
		this.menuExcluirAbaAtiva2 = new MenuItem("Excluir aba ativa");
		this.menuExcluirAbaAtiva2.setName("menuExcluirAbaAtiva2");
		this.menuExcluirAbaAtiva2.setCanFocus(false);
	
		this.menuAlterarNomeAbaAtiva2 = new MenuItem("Alterar nome da aba ativa");
		this.menuAlterarNomeAbaAtiva2.setName("menuAlterarNomeAbaAtiva2");
		this.menuAlterarNomeAbaAtiva2.setCanFocus(false);
	
		this.menuIncluirNovaAba2 = new MenuItem("Incluir nova aba");
		this.menuIncluirNovaAba2.setName("menuIncluirNovaAba2");
		this.menuIncluirNovaAba2.setCanFocus(false);
	
		this.separador2 = new SeparatorMenuItem();
		this.separador2.setName("separador2");
		this.separador2.setCanFocus(false);
	
		this.menuColapsarDiretorios2 = new MenuItem("Colapsar Diretórios");
		this.menuColapsarDiretorios2.setName("menuColapsarDiretorios2");
		this.menuColapsarDiretorios2.setCanFocus(false);
	
		this.menuExpandirDiretorios2 = new MenuItem("Expandir Diretórios");
		this.menuExpandirDiretorios2.setName("menuExpandirDiretorios2");
		this.menuExpandirDiretorios2.setCanFocus(false);
	
		this.menuExcluirDiretorioSelecionado = new MenuItem("Excluir Diretório Selecionado");
		this.menuExcluirDiretorioSelecionado.setName("menuExcluirDiretorioSelecionado");
		this.menuExcluirDiretorioSelecionado.setCanFocus(false);
	
		this.menuInformacoesDiretorioArquivo = new MenuItem("Informações do Diretório / Arquivo");
		this.menuInformacoesDiretorioArquivo.setName("menuInformacoesDiretorioArquivo");
		this.menuInformacoesDiretorioArquivo.setCanFocus(false);
	
		this.setName("menuPopup");
		this.setCanFocus(false);
		this.add(menuInformacoesDiretorioArquivo);
		this.showAll();
		this.add(menuExcluirDiretorioSelecionado);
		this.showAll();
		this.add(menuExpandirDiretorios2);
		this.showAll();
		this.add(menuColapsarDiretorios2);
		this.showAll();
		this.add(separador2);
		this.showAll();
		this.add(menuIncluirNovaAba2);
		this.showAll();
		this.add(menuAlterarNomeAbaAtiva2);
		this.showAll();
		this.add(menuExcluirAbaAtiva2);
		this.showAll();
	

		this.menuInformacoesDiretorioArquivo.connect(new MenuItem.Activate(){
			@Override
			public void onActivate(MenuItem mi){
				on_menuInformacoesDiretorioArquivo_activate(mi);
			}
		});
		this.menuExcluirDiretorioSelecionado.connect(new MenuItem.Activate(){
			@Override
			public void onActivate(MenuItem mi){
				on_menuExcluirDiretorioSelecionado_activate(mi);
			}
		});
		this.menuExpandirDiretorios2.connect(new MenuItem.Activate(){
			@Override
			public void onActivate(MenuItem mi){
				on_menuExpandirDiretorios2_activate(mi);
			}
		});
		this.menuColapsarDiretorios2.connect(new MenuItem.Activate(){
			@Override
			public void onActivate(MenuItem mi){
				on_menuColapsarDiretorios2_activate(mi);
			}
		});
		this.menuIncluirNovaAba2.connect(new MenuItem.Activate(){
			@Override
			public void onActivate(MenuItem mi){
				on_menuIncluirNovaAba2_activate(mi);
			}
		});
		this.menuAlterarNomeAbaAtiva2.connect(new MenuItem.Activate(){
			@Override
			public void onActivate(MenuItem mi){
				on_menuAlterarNomeAbaAtiva2_activate(mi);
			}
		});
		this.menuExcluirAbaAtiva2.connect(new MenuItem.Activate(){
			@Override
			public void onActivate(MenuItem mi){
				on_menuExcluirAbaAtiva2_activate(mi);
			}
		});

	}

	public static void mostrar() 
	{
		
	}

	public void on_menuInformacoesDiretorioArquivo_activate(MenuItem mi){

	}

	public void on_menuExcluirDiretorioSelecionado_activate(MenuItem mi){
		
	}

	public void on_menuExpandirDiretorios2_activate(MenuItem mi){
		
	}

	public void on_menuColapsarDiretorios2_activate(MenuItem mi){
		
	}

	public void on_menuIncluirNovaAba2_activate(MenuItem mi){
		
	}

	public void on_menuAlterarNomeAbaAtiva2_activate(MenuItem mi){
		
	}

	public void on_menuExcluirAbaAtiva2_activate(MenuItem mi){
		
	}


}
